nncf/quantization/algorithms/min_max/algorithm.py

```diff
@@ -716,3 +719,87 @@ def _apply_model_type_pass(
                                 f"Update quantization mode for the node {node_name}"
                                 f" to the symmetric due to ModelType parameter."
                             )
+
+    def _apply_device_pass(self, target_device: TargetDevice, quantized_model: TModel) -> TModel:
+        """
+        This method applies model post-processing device passes.
+
+        :param target_device: TargetDevice instance.
+        :param quantized_model: Quantized TModel instance.
+        :return: Modified TModel.
+        """
+
+        passes_map = {TargetDevice.CPU_SPR: self._apply_spr_pass}
+
+        if target_device not in passes_map:
+            return quantized_model
+
+        return passes_map[target_device](quantized_model)
+
+    def _apply_spr_pass(self, quantized_model: TModel) -> TModel:
+        """
+        Applies CPU_SPR-related pass.
+        The main action is to remove one of the quantizers before elementwise layer (e.g. Add).
+        This action allows to get performance boost on SPR devices.
+
+        :param quantized_model: Quantized TModel instance.
+        :return: Modified TModel.
+        """
+
+        def _is_node_after_producers(node):
+            input_node = node
+            while True:
+                input_node = nncf_graph.get_previous_nodes(input_node)
+                if len(input_node) > 1:
+                    return False
+                input_node = input_node[0]
+                if input_node.metatype in producer_metatypes:
+                    return True
+
+        nncf_graph = NNCFGraphFactory.create(quantized_model)
```

Comment:
It significantly slow down the minmax algorithm for CPU_SPR device. I would recommend to think about implement this pass on the original graph with the quantization setup.

Reply:
Changed.